### PR TITLE
Allow customers to specify $(CppWinRTVersion)

### LINKED ISSUE #10794: Allow user to override CppWinRT version
### Summary

Allow user to override the default version similar to how RNW handles WinUI version.

### Motivation

When RNW is consumed with other packages there may be a mismatch in CppWinRT version.

### Basic Example

_No response_

### Open Questions

_No response_

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

This PR updates our shared prop file where we set the version of
$(CppWinRTVersion) to use to only specify it if it's not already
specifed by the customer (say in ExperimentalFeatures.props).

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Some customers have complained that they want to use a different version of CppWinRT that what we specify.

Closes #10794

### What

Added a null-check when we set the version.

## Screenshots
N/A

## Testing
N/A

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10793)

### CODE REVIEW

**Comment by jonthysell** [comment]
This will need to be backported to RNW >= 0.69.

File: vnext/PropertySheets/External/Microsoft.ReactNative.Cpp.Dependencies.props [hunk 1]
```diff
@@ -11,7 +11,7 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <PropertyGroup Label="CppWinRT">
-    <CppWinRTVersion>2.0.211028.7</CppWinRTVersion>
+    <CppWinRTVersion Condition="'$(CppWinRTVersion)' == ''">2.0.211028.7</CppWinRTVersion>
```
Comment (JunielKatarn) [inline vnext/PropertySheets/External/Microsoft.ReactNative.Cpp.Dependencies.props:14]: We should probably validate a minimum version.
Versions earlier than this default have a packaging problem which made `PackageReference` fail.

**Review by JunielKatarn — APPROVED** [review]
Wondering if we should add a prefix for consumer-facing MSBuild properties.
For example, `$(RnwCppWinRTVersion)`.

**Comment by slobo80** [comment]
I just opened an issue for this https://github.com/microsoft/react-native-windows/issues/10794

**Review by vmoroz — APPROVED** [review]

**Pull request merged**